RFS/src/main/java/com/rfs/common/SnapshotShardUnpacker.java

```diff
@@ -58,6 +72,36 @@ public void unpack(ShardMetadata.Data shardMetadata) {
         }
     }
 
+    @Override
+    public void close() {
+        try {
+            Path luceneIndexDir = Paths.get(luceneFilesBasePath + "/" + shardMetadata.getIndexName() + "/" + shardMetadata.getShardId());
+            if (Files.exists(luceneIndexDir)) {
+                deleteRecursively(luceneIndexDir);
+            }
+            
+        } catch (Exception e) {
+            throw new CouldNotCleanUpShard("Could not clean up shard: Index " + shardMetadata.getIndexId() + ", Shard " + shardMetadata.getShardId(), e);
```

Comment:
Should/does this kill the process?  Seems like it probably should since it would be safer to recycle the process, especially since you were at the end of working (if you have other pending work, maybe it should not take more on and flush out the current work items).

Reply:
Makes sense.  With the way things are currently implemented, I believe it will kill the process.

Reply:
Do you have a test to confirm that?  I ask because making sure that processes die, especially through refactorings, can be pretty tricky to maintain.